### PR TITLE
Allow the install playbook to run from either your local machine or 

### PR DESCRIPTION
**Pull request by akrzos**
orchestration machine

### CODE REVIEW

**Comment by akrzos** [comment]
@chaitanyaenr @wabouhamad Please review and merge after you test.  The gist here is we can run the playbook from more than one location now.

For Example:

akrzos's orchestration machine in the cloud:
```
root@ip-172-31-32-21: ~/scale-ci-ansible # head -n 3 inventory 
[orchestration]
# The server that contains the kubeconfig and oc command
localhost
root@ip-172-31-32-21: ~/scale-ci-ansible # time ansible-playbook -vv -i inventory OCP-4.X/install.yml
```

akrzos's laptop but using the orchestration machine in the cloud:
```
[akrzos@bithead scale-ci-ansible]$ head -n 3 inventory
[orchestration]
# The server that contains the kubeconfig and oc command
54.X.X.X 
[akrzos@bithead scale-ci-ansible]$ time ansible-playbook -vv -i inventory OCP-4.X/install.yml
```

In both cases the playbook can run successfully now.  In case two, I can now edit the playbook and use `git` commands to propose changes from my local machine, making it much more convenient to make changes..  Case 2 running would also simulate being the Jenkins server.

File: OCP-4.X/roles/controller-kickstart/tasks/main.yml [hunk 1]
```diff
@@ -9,21 +9,21 @@
...
     state: directory
   with_items:
     - /home/core/.kube
     - /root/.kube
- 
+
 - name: copy the kubeconfig to controller node
   copy:
-    src: "{{ workdir }}/auth/kubeconfig"
+    src: "{{playbook_dir}}/kubeconfig"
```
Comment (wabouhamad) [inline OCP-4.X/roles/controller-kickstart/tasks/main.yml]: In the case where KUBECONFIG_AUTH_DIR_PATH inventory variable is defined and has a value, is the "workdir" variable being used in this controller-kickstart role task file ?  Where do we define the "playbook_dir" variable ?  Is it in the inventory file ?

Comment (akrzos) [reply OCP-4.X/roles/controller-kickstart/tasks/main.yml]: The `playbook_dir` variable is a [magic variable](https://docs.ansible.com/ansible/latest/user_guide/playbooks_variables.html#accessing-information-about-other-hosts-with-magic-variables) by Ansible.  In the above sense it provides the full path to a place to store the kubeconfig.

Example:
```
[akrzos@bithead scale-ci-ansible]$ ansible-playbook -vv -i inventory OCP-4.X/install.yml
```

`playbook_dir` for the above `ansible-playbook` run would result in the full path to directory `scale-ci-ansible/OCP-4.X` (Wherever scale-ci-ansible is cloned to)

Comment (wabouhamad) [reply OCP-4.X/roles/controller-kickstart/tasks/main.yml]: Flexy creates an archive zip file containing the kubeconfig of the cluster it just installed.  We have a job to download that zip archive and expand it on jump host.  In the inventory file, we specify the full path to the auth dir containing the kubeconfig file in variable "KUBECONFIG_AUTH_DIR_PATH", in the expanded zip archive. 

Proposed change:

- name: copy the kubeconfig to controller node
  copy:
    src: "{{playbook_dir}}/kubeconfig"
    dest: "{{ item }}"
  with_items:
    - /home/core/.kube/config
    - /root/.kube/config
  when:
    - KUBECONFIG_AUTH_DIR_PATH is undefined

# handle case where kubeconfig is in another specified dir
- name: copy the kubeconfig to controller node from specific directory
  copy:
    src: "{{workdir}}/auth/kubeconfig"
    dest: "{{ item }}"
  with_items:
    - /home/core/.kube/config
    - /root/.kube/config
  when:
    - KUBECONFIG_AUTH_DIR_PATH is defined
    - KUBECONFIG_AUTH_DIR_PATH != ""

Comment (akrzos) [reply OCP-4.X/roles/controller-kickstart/tasks/main.yml]: I believe I have addressed this.  I'll try out the install again to make sure this works in the cases mentioned.  Please re-review and let me know if this won't work for flexy.

**Comment by akrzos** [comment]
@wabouhamad can you test again with Flexy?

**Comment by wabouhamad** [comment]
@akrzos I am getting one failure at task in role post-config: 
OCP-4.X/roles/post-config/tasks/main.yml:

```name: Get the kubeconfig off of the orchestration host
      fetch:
        src: "{{ GOPATH }}/src/github.com/openshift/installer/auth/kubeconfig"
        dest: "{{playbook_dir}}/kubeconfig"
        flat: true
```
we would need to handle both cases again here:
```
when:
    - KUBECONFIG_AUTH_DIR_PATH is undefined
```
and 
```
  when:
    - KUBECONFIG_AUTH_DIR_PATH is defined
    - KUBECONFIG_AUTH_DIR_PATH != ""
```

**Comment by akrzos** [comment]
@wabouhamad Are we good to merge this now?

**Pull request merged**